### PR TITLE
Debug mode - Issue/4864

### PR DESCRIPTION
**Pull request by pippinsplugins**
#4864

### CODE REVIEW

File: includes/admin/settings/register-settings.php [hunk 1]
```diff
@@ -668,6 +668,12 @@ function edd_get_registered_settings() {
 						'desc' => sprintf(__('Allow quantities to be adjusted when adding %s to the cart, and while viewing the checkout cart.','easy-digital-downloads' ), edd_get_label_plural( true ) ),
 						'type' => 'checkbox',
 					),
+					'debug_mode' => array(
+						'id'   => 'debug_mode',
+						'name' => __( 'Debug Mode', 'easy-digital-downloads' ),
+						'desc' => __( 'Check this box to enable debug mode. When enabled, debug messages will be logged and shown in Downloads &rarr; Tools &rarr; Debug Mode.', 'easy-digital-downloads' ),
```
Comment (sunnyratilal) [inline includes/admin/settings/register-settings.php]: Needs changing to `Downloads &rarr; Tools &rarr; Debug Log`

**Comment by sunnyratilal** [comment]
@pippinsplugins Is it necessary to have [this](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/88d6dfcbb1799f267778b2086e45d122af53aa5e/includes/class-edd-logging.php#L483). Instead, could it be added as an object to the `Easy_Digital_Downloads` class like we do with our other classes?  Just so it keeps everything consolidated?

**Comment by sunnyratilal** [comment]
Besides my above comments, I've tested the functionality and it seems to be working well 👍

**Comment by pippinsplugins** [comment]
@sunnyratilal that global has been there since _way_ back and is unrelated to this change so I'd rather we not change it.

**Comment by chriscct7** [comment]
It'd be nice if edd_is_debug_mode could be additionally turned on via a constant

```
$ret = edd_get_option( 'debug_mode', false );
$ret = $ret || ( is_defined( 'EDD_DEBUG_MODE' ) && EDD_DEBUG_MODE );
```

**Comment by chriscct7** [comment]
Personally, re:global, I'd remove it like sunny suggested (self::$instance->logs = new EDD_Logging(); ) and then in the bottom of EDD main file do:

$GLOBALS['edd_logs'] = EDD()->logs;

That way you can use the logs off the global object (expected) and if you have a person editing the log object it will also affect the global in EDD so that other plugins don't have unexpected results.

That's fully backwards compatible, and is a lot more expected

**Pull request merged**